filters/diag/diag.go

```diff
@@ -417,3 +451,89 @@ func (t *throttle) Response(ctx filters.FilterContext) {
 	rsp := ctx.Response()
 	rsp.Body = t.goThrottle(rsp.Body, true)
 }
+
+func (j *jitter) Name() string {
+	switch j.typ {
+	case normalDist:
+		return "jitterNormal"
+	case uniformDist:
+		return "jitterUniform"
+	}
+	return "unknown"
+}
+
+func (j *jitter) CreateFilter(args []interface{}) (filters.Filter, error) {
+	var (
+		delay     time.Duration
+		interval  time.Duration
+		mu, sigma float64
+		ok        bool
+		err       error
+	)
+
+	if !(len(args) == 2 || len(args) == 4) {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	if delay, err = parseDuration(args[0]); err != nil {
+		return nil, fmt.Errorf("failed to parse duration delay %v: %w", args[0], err)
+	}
+
+	if interval, err = parseDuration(args[1]); err != nil {
+		return nil, fmt.Errorf("failed to parse duration interval %v: %w", args[1], err)
+	}
+
+	switch j.typ {
+	case normalDist:
+		switch len(args) {
+		case 2:
+			mu = 0
+			sigma = 1
+		case 4:
+
+			mu, ok = args[2].(float64)
+			if !ok {
+				return nil, fmt.Errorf("failed to convert mu to float64 %v", args[2])
+			}
+			sigma, ok = args[3].(float64)
+			if !ok {
+				return nil, fmt.Errorf("failed to convert sigma to float64 %v", args[3])
+			}
+		}
+	case uniformDist:
+		if len(args) != 2 {
+			return nil, filters.ErrInvalidFilterParameters
+		}
+	default:
+		return nil, errors.New("distribution does not exist")
+	}
+
+	return &jitter{
+		typ:      j.typ,
+		delay:    delay,
+		interval: interval,
+		mu:       mu,
+		sigma:    sigma,
+	}, nil
+}
+
+func (j *jitter) Request(filters.FilterContext) {
+	d := j.delay
+
+	switch j.typ {
+	case uniformDist:
+		/* #nosec */
+		r := rand.Float64() - 0.5 // +/- \epsilon
+		f := r * float64(j.interval)
+		d += time.Duration(int64(f))
+	case normalDist:
+		r := rand.NormFloat64()
+		r = r*j.sigma + j.mu
+		f := r * float64(j.interval)
+		d += time.Duration(int64(f))
```

Comment:
Lets check this, I think it should be
```go
d = rand.NormFloat64()*j.sigma + j.delay
```
i.e. mu is the mean or expectation of the distribution which is exactly the delay

Reply:
so you mean to drop j.mu, right?

Reply:
https://en.wikipedia.org/wiki/Standard_deviation
> A useful property of the standard deviation is that unlike the variance, it is expressed in the same unit as the data.

> If a data distribution is approximately normal then about 68 percent of the data values are within one standard deviation of the mean (mathematically, μ ± σ, where μ is the arithmetic mean), about 95 percent are within two standard deviations (μ ± 2σ), and about 99.7 percent lie within three standard deviations (μ ± 3σ). This is known as the 68-95-99.7 rule, or the empirical rule.

I think "normal" filter should have two  `time.Duration` parameters mean and sigma.

Reply:
One problem is that I don't have +/-, only + in the normalDist case.

Reply:
I used float64 because of https://golang.org/pkg/math/rand/#NormFloat64, but maybe it's better as you wrote.
Let me test.

Reply:
I changed the code to have only 2 params for normalJitter() reusing the var from uniformJitter.